.github/workflows/pre-build-validation.yml

```diff
@@ -27,7 +33,38 @@ jobs:
         repository: ${{ github.event.inputs.owner }}/${{ github.event.inputs.repo }}
         ref: ${{ github.event.inputs.branch }}
         path: repo
+        fetch-depth: 0
+    - name: Check for Product Changes
+      id: pull_request
+      run: |
+        cd repo
+        yamlfiles=$(git diff --name-only origin/main -- mmv1/products)
+        if [ -z "$yamlfiles" ]; then
+          echo "has_changes=false" >> $GITHUB_OUTPUT
+        else
+          echo "has_changes=true" >> $GITHUB_OUTPUT
+        fi
+    - name: Get Job URL
+      if: ${{ !failure() && steps.pull_request.outputs.has_changes == 'true' }}
+      id: get_job
+      run: |
+        response=$(curl --get -Ss -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" -H "Accept: application/vnd.github.v3+json" "https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/attempts/${{ github.run_attempt }}/jobs")
+        html_url=$(echo "$response" | jq -r --arg job_name "${{ github.job }}" '.jobs | map(select(.name == $job_name)) | .[0].html_url')
+        echo "url=${html_url}" >> $GITHUB_OUTPUT
     - name: Install yamllint
+      if: ${{ !failure() && steps.pull_request.outputs.has_changes == 'true' }}
       run: pip install yamllint
     - name: Lint YAML files
-      run: yamllint mmv1/products
+      if: ${{ !failure() && steps.pull_request.outputs.has_changes == 'true' }}
+      run: yamllint -c repo/.yamllint repo/mmv1/products
+    - name: Post Result Status to Pull Request
+      if: ${{ !cancelled() && steps.pull_request.outputs.has_changes == 'true' }}
+      run: |
+        curl -X POST -H "Authorization: token ${{secrets.GITHUB_TOKEN}}" \
+        -H "Accept: application/vnd.github.v3+json" \
+        "https://api.github.com/repos/GoogleCloudPlatform/magic-modules/statuses/${{github.event.inputs.sha}}" \
+        -d '{
+        "context": "${{ github.event.inputs.repo }}${{ env.status_suffix }}",
```

Comment:
status_suffix doesn't appear to be set here in the workflow. 

See https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.github/workflows/test-tpg.yml#L8-L9